### PR TITLE
Really handle notCurrentLeaderExceptions

### PR DESCRIPTION
**Pull request by sudiksha27**
**Goals (and why)**:
We actually want to handle the exceptions while executing batched requests.

**Testing (What was existing testing like?  What have you done to improve it?)**:
TBD

**Priority (whenever / two weeks / yesterday)**:
Yesterday 🔥

### CODE REVIEW

**Review by jeremyk-91 — COMMENTED** [review]
🕐 0:05 🏁 ????

This looks broadly fine, though there seem to be some tests failing on container 6 (and I think this is a bug in the existing test, to be clear, not in your impl) that we'll need to fix.

File: timelock-impl/src/test/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResourceTest.java [hunk 1]
```diff
@@ -105,12 +108,23 @@ public void canGetLeaderTimesForMultipleClients() {
...
-                .isInstanceOf(BlockingTimeoutException.class);
+                .hasCauseInstanceOf(Throttle.class);
+    }
+
+    @Test
+    public void handlesNotCurrentLeaderExceptions() {
+        String throwingClient = "alpha";
+        Set<Namespace> namespaces = ImmutableSet.of(Namespace.of(throwingClient), Namespace.of("beta"));
+        when(getServiceForClient(throwingClient).leaderTime())
+                .thenThrow(new NotCurrentLeaderException("Not the " + "leader!"));
```
Comment (Jolyon-S) [inline timelock-impl/src/test/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResourceTest.java:124]: nice checkstyle

**Review by Jolyon-S — APPROVED** [review]
Nice! Just fix up that string.

**Comment by Jolyon-S** [comment]
oh yeah and fix the test

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockServer.java [hunk 2]
```diff
@@ -140,6 +140,11 @@ TestableLeaderPinger pinger() {
         throw new SafeIllegalStateException("unexpected mode", SafeArg.of("mode", mode));
     }
 
+    public boolean isMultiLeader() {
+        PaxosLeaderMode mode = serverHolder.installConfig().paxos().leaderMode();
+        return mode == PaxosLeaderMode.LEADER_PER_CLIENT ? true : false;
```
Comment (jeremyk-91) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockServer.java:145]:
```suggestion
        return mode == PaxosLeaderMode.LEADER_PER_CLIENT;
```

**Review by jeremyk-91 — APPROVED** [review]

**Pull request merged**